### PR TITLE
Check if ICE issues are closed upstream and create issues to triage them

### LINKED ISSUE #613: Detect if an issue is closed
Glacier doen't report anything even if an issue is already closed. It should be, however, since we have to triage it and deal with that ICE (remove, change issue number, re-open, etc.).

### PR DESCRIPTION
**Pull request by mibac138**
Fixes #613
Example issue: mibac138/triagebot-test#5
As long as an issue with label `triage` and title starting with `Issue {id}` exists, new issues won't be created about the same ICE.

### CODE REVIEW

File: labeler/src/main.rs
Comment (JohnTitor) [inline labeler/src/main.rs:75]: I think it's better just to rename `i` to `id`.

File: labeler/src/main.rs
Comment (JohnTitor) [inline labeler/src/main.rs:79]:
```suggestion
            &format!("issue-{} has been closed", i),
```

I'd prefer this format.

**Review by JohnTitor — APPROVED** [review]
Overall LGTM with some nits.

**Comment by mibac138** [comment]
@JohnTitor thanks for the quick review, I applied your suggestions

**Comment by JohnTitor** [comment]
Thanks!

**Comment by JohnTitor** [comment]
The creates the list:
```
closed issues list: [84434, 84408, 84399, 84149, 84148, 84108, 84104, 83907, 83893, 83836, 83753, 83737, 83691, 83641, 83621, 83613, 83499, 83494, 83466, 83340, 83309, 83253, 83182, 83176, 83175, 83138, 83048, 83017, 82981, 82952, 82869, 82865, 82835, 82792, 82772, 82610, 82455, 82438, 82418, 82329, 82326, 82325, 82290, 82156, 82126, 81924, 81845, 81806, 81800, 81712, 81246, 81208, 81193, 81006, 80779, 80767, 80742, 80664, 80561, 80531, 80433, 80371, 80231, 80230, 80229, 80228, 80207, 80134, 80077, 80062, 80060, 79949, 79787, 79768, 79690, 79669, 79666, 79636, 79569, 79566, 79537, 79518, 79422, 78922, 78806, 78722, 78721, 78671, 78653, 78652, 78651, 78456, 78246, 77993, 77919, 77911, 77910, 77475, 77329, 77323, 77320, 77218, 76535, 76375, 76202, 76064, 76013, 75983, 75962, 75883, 75707, 75682, 75415, 75323, 75299, 75229, 75158, 75153, 75053, 74918, 74906, 74816, 74761, 74702, 74634, 74608, 74599, 74447, 74358, 74307, 74298, 74255, 74244, 74199, 74047, 74018, 73993, 73980, 73860, 73730, 73626, 73541, 73375, 73327, 73260, 73252, 73249, 73229, 73109, 73035, 72911, 72819, 72680, 72679, 72565, 72554, 72651, 72372, 72352, 72285, 72278, 72267, 72181, 72113, 72105, 72076, 71922, 71805, 71793, 71699, 71659, 71611, 71516, 71471, 71443, 71382, 71381, 71348, 71344, 71316, 71297, 71212, 71202, 71176, 71169, 71042, 71036, 70972, 70934, 70877, 70746, 70723, 70624, 70596, 70586, 70507, 70381, 70304, 70303, 70292, 70217, 70121, 70114, 69970, 69913, 69840, 69654, 69490, 69415, 69414, 69401, 69396, 69378, 69307, 69296, 69239, 69204, 69191, 69136, 69130, 69092, 68890, 68868, 68841, 68684, 68656, 68653, 68652, 68650, 68649, 68648, 68645, 68644, 68643, 68621, 68596, 68578, 68550, 68542, 68532, 68398, 68396, 68394, 68393, 68368, 68296, 68205, 68136, 68130, 68092, 68091, 68060, 68025, 68013, 67883, 67858, 67856, 67844, 67739, 67558, 67552, 67514, 67424, 67405, 67377, 67375, 67166, 67158, 67019, 66962, 66930, 66906, 66706, 66696, 66695, 66693, 66580, 66501, 66500, 66473, 66387, 66371, 66357, 66340, 66270, 66223, 66205, 65934, 65918, 65774, 65679, 65673, 65581, 65561, 65466, 65462, 65394, 65383, 65348, 65159, 64848, 64755, 64537, 63900, 63695, 63322, 63300, 63279, 63263, 63154, 63116, 62879, 62767, 62708, 62579, 62524, 62521, 62504, 62480, 62456, 62222, 62200, 62046, 61936, 61774, 61747, 61455, 61368, 61188, 60814, 60619, 60473, 59996, 59494, 59435, 59311, 59191, 59184, 58987, 58094, 57739, 57201, 57200, 56445, 56229, 55414, 54108, 54038, 53448, 52843, 52808, 51892, 51506, 51173, 49362, 44861, 43924, 41073, 31437, 30606, 29382, 22565, 21599, 15402, 13520, 13368]
```

Well, we haven't checked if it exists on `./ices`. And CI fails on unwrap: https://github.com/rust-lang/glacier/runs/2768684558

**Comment by mibac138** [comment]
I'm pretty sure you need to create a `triage` label and add it to the newly created issues or else the bot will create duplicate issues; if you'd prefer I can change it to search all issues, not only those labeled triage for closed-issue issues. However I don't really understand why this fails. `403` means forbidden, but the request wasn't blocked when creating the previous issues, so I'm not sure why github doesn't like it.

**Comment by JohnTitor** [comment]
I created the label before merging but it failed, we should have reached the rate limit, yeah.
So, the current `closed_issue_numbers` contains all the closed issues on rust-lang/rust but we haven't checked whether it's on `ices/`, right? I've opened #791, this should fix that issue.

**Comment by mibac138** [comment]
Yeah, that's right. #791 looks right to me. Regarding rate limiting something like https://github.com/rust-lang/triagebot/blob/master/src/github.rs#L22-L147 should work but I wonder if it isn't overkill, since this situation should solve itself after a few runs. It's unlikely for over 20 glacier tracked issues to be closed on a single day, right?

**Comment by JohnTitor** [comment]
> Regarding rate limiting something like https://github.com/rust-lang/triagebot/blob/master/src/github.rs#L22-L147 should work but I wonder if it isn't overkill, since this situation should solve itself after a few runs. It's unlikely for over 20 glacier tracked issues to be closed on a single day, right?

Yes, I think so, unless [it's the first time](https://github.com/rust-lang/glacier/runs/2768936545?check_suite_focus=true). If it happens often, we can port the code anytime.

**Comment by mibac138** [comment]
It looks like the labels weren't added once again. I'm really confused why that is. The test run in my repo (https://github.com/mibac138/triagebot-test/issues/5) did add a triage label which makes the situation quite odd. I'm sorry for the hassle this is causing

**Comment by JohnTitor** [comment]
I guess adding a label is executed after opening all the issues. If not, it should've failed when it created the first issue. That also explains why the opening issues don't have a label.

**Comment by JohnTitor** [comment]
> 
> 
> I guess adding a label is executed after opening all the issues. If not, it should've failed when it created the first issue. That also explains why the opening issues don't have a label.

Hmm, this wasn't correct, now CI is green but the issues still don't have a label. Yeah, it's quite odd.

**Comment by mibac138** [comment]
Actually, could it be because the bot doesn't have push permissions? See the [GitHub API docs](https://docs.github.com/en/rest/reference/issues#create-an-issue) note for labels: "NOTE: Only users with push access can set labels for new issues. Labels are silently dropped otherwise."

**Comment by JohnTitor** [comment]
Ah, I guess so! @rust-lang-glacier-bot should have the access to rust-lang/rust but IIRC I didn't ask the infra team to grant the access here.

**Pull request merged**
